src/main/java/com/twilio/security/RequestValidator.java

```diff
@@ -107,4 +113,23 @@ private boolean secureCompare(String a, String b) {
         return mismatch == 0;
     }
 
+    private String removePort(String url) {
+        try {
+            URI parsedUrl = new URI(url);
+            return (parsedUrl.getPort()) == -1 ? url : url.replaceFirst(":\\d+", "");
+        } catch (Exception e) {
+            return "";
+        }
+    }
+    
+    private String addPort(String url) {
+        try {
+            URI parsedUrl = new URI(url);
+            String port = Objects.equals(parsedUrl.getScheme(), "https") ? "443" : "80";
+            return (parsedUrl.getPort()) != -1 ? url : url.replace(".com", ".com:" + port);
```

Comment:
Is there a URI builder in Java that can help here? Something like `https://www.complete.com` would not turn out well.

Reply:
fixed. Thanks for pointing that out!